hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto

```diff
@@ -1027,6 +1027,7 @@ message OpenKeyBucket {
 message OpenKey {
   required string name = 1;
   required uint64 clientID = 2;
+  optional uint64 parentID = 3;
```

Comment:
For the same reason, maybe we shouldn't store `client ID` either?

Reply:
I believe this will work. The bucket layout should be set based on whether the key was retrieved from the open key table or open file table, since the bucket itself may no longer exist. The field will be used by followers to know which table they need to move the key from.

It is interesting that FSO has its own table for open keys, but looks to share the same delete table as OBS and LEGACY keys. I would think there would either be different tables for both or they would both use the same table. Either way, by having this bucket layout field we could handle a delete table unique to FSO in the future. cc @rakeshadr to inquire why the tables are used in this way out of curiosity, although it does not really affect this patch.

Reply:
Since removing a field in proto is hard, I decide not to add `BucketLayout` for now.
Because the caller must know which bucketLayout it has passed to `getExpiredOpenKeys`.
Maybe we should add bucketLayout in `OpenKeysDeleteRequest` instead.

Reply:
@errose28 Thanks for bringing up this point.
>It is interesting that FSO has its own table for open keys, but looks to share the same delete table as OBS and LEGACY keys. I would think there would either be different tables for both or they would both use the same table.

Following are the bucket layout FSO feature specific tables. Since KeyDeletingService uses an offline table which handles deleted files and its physical block cleanup, we thought of making use of the existing logic with minimal efforts for the cleanup and again there was no requirement to introduce a logic based on specific bucket layout types. I am open to introduce a FSO specific table if there is a strong requirement.

```
* FSO/Prefix Tables:
* |----------------------------------------------------------------------|
* |  directoryTable    | parentId/directoryName -> DirectoryInfo         | 
* |----------------------------------------------------------------------|
* |  fileTable         | parentId/fileName -> KeyInfo                    |
* |----------------------------------------------------------------------|
* |  openFileTable     | parentId/fileName/id -> KeyInfo                 |
* |----------------------------------------------------------------------|
* |  deletedDirTable   | parentId/directoryName -> KeyInfo    
* |----------------------------------------------------------------------|
```